src/vanilla.ts

```diff
@@ -73,7 +73,7 @@ const createStoreImpl: CreateStoreImpl = (createState) => {
     if (!Object.is(nextState, state)) {
       const previousState = state
       state =
-        replace ?? typeof nextState !== 'object'
+        replace ?? ( typeof nextState !== 'object' || nextState === null )
```

Comment:
I would use `Object.prototype.toString` since typeof could return `object` in many cases. Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof

```suggestion
        replace ?? Object.prototype.toString.call(nextState).slice(8, -1) === 'Object'
```

Reply:
Sorry, but I don't prefer that. typeof should be good for our purpose and for edge cases, users can use `replace`.

Reply:
@dai-shi in that case we need to add a new test case for this change

Reply:
In any case, a test would be good. @wulimao49 would you be able to add one?

Wait, on second thought, this isn't an issue with `null` specifically.
Maybe, it's more reasonable to do this: `typeof state !== 'object' || state === null`
What do you guys think?

Reply:
`typeof state !== 'object' || state === null` will make it impossible to set a object state to `null` except explicitly set `replace` to `true`.

both do it on `state` and `nextState` will have some kind of edge cases.

maybe we should make the merging logic configable? or just using `replace` is enough.

 yes i can set a test if we decide a certain solution on this.

Reply:
Here's my decision. This doesn't cover edge cases. It's intentional.
Just pick either one please:
- `typeof nextState !== 'object' || nextState === null`
- `typeof state !== 'object' || state === null`

Which makes less error-prone?

Reply:
@dai-shi in that case I'll go with the second one

Reply:
@wulimao49 How about you?

Reply:
@dai-shi i prefer the second one

Reply:
Let's go with `typeof state !== 'object' || state === null`.
Can you create some test cases?

Reply:
test cases added

Reply:
Yeah, that's too much. 99% users don't use this capability anyway.

I considered changing to `replace ?? defaultReplace`, and `const defalutReplace = typeof state !== 'object || state === null` on initialization. But, its capability is rather limited, and the behavior can be less predictable.

I think the current solution is a fair point.